homeassistant/components/camera/uvc.py

```diff
@@ -0,0 +1,96 @@
+"""
+Ubiquiti Unifi Video Cameras
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Ubiquiti's UVC cameras
+
+Configuration:
+
+camera:
+  platform: uvc
+  nvr: ADDRESS
+  port: PORT
+  key: APIKEY
+
+Variable:
+
+nvr
+*Required
+ADDRESS should be the IP or hostname of the NVR server
+
+port
+*Optional
+PORT is the port number to use for accessing the NVR
+
+key
+*Required
+APIKEY should be the API key available from the NVR web interface
+"""
+import logging
+import socket
+
+from homeassistant.helpers import validate_config
+from homeassistant.components.camera import DOMAIN, Camera
+
+REQUIREMENTS = ['uvcclient>=0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Discover cameras on a Unifi NVR. """
+    if not validate_config({DOMAIN: config}, {DOMAIN: ['nvr', 'key']},
+                           _LOGGER):
+        return None
+
+    addr = config.get('nvr')
+    port = int(config.get('port', 7080))
+    key = config.get('key')
+
+    from uvcclient import nvr
+    nvrconn = nvr.UVCRemote(addr, port, key)
```

Comment:
Hmm, not very gracefully (my fault in the library).

I can do something and check for the TypeError it will throw and then fix it more cleanly in a later patch after I augment uvcclient.

Preference?